linux_factory.go

```diff
@@ -163,16 +214,346 @@ func (l *linuxFactory) loadContainerState(root string) (*configs.State, error) {
 	return state, nil
 }
 
-// StartInitialization loads a container by opening the pipe fd from the parent to read the configuration and state
-// This is a low level implementation detail of the reexec and should not be consumed externally
-func (f *linuxFactory) StartInitialization(pipefd uintptr) (err error) {
-	pipe := os.NewFile(uintptr(pipefd), "pipe")
+func (l *linuxFactory) validateID(id string) error {
+	if !idRegex.MatchString(id) {
+		return newGenericError(fmt.Errorf("Invalid id format: %v", id), InvalidIdFormat)
+	}
+	if len(id) > maxIdLen {
+		return newGenericError(fmt.Errorf("Invalid id format: %v", id), InvalidIdFormat)
+	}
+	return nil
+}
 
-	setupUserns := os.Getenv("_LIBCONTAINER_USERNS")
-	pid := os.Getenv("_LIBCONTAINER_INITPID")
-	if pid != "" && setupUserns == "" {
-		return namespaces.InitIn(pipe)
+func (l *linuxFactory) initDefault(uncleanRootfs string, process *processArgs) (err error) {
+	config := process.Config
+	networkState := process.NetworkState
+
+	// TODO: move to validation
+	/*
+		rootfs, err := utils.ResolveRootfs(uncleanRootfs)
+		if err != nil {
+			return err
+		}
+	*/
```

Comment:
I was thinking we can do it in another pr at the `factory.Create()` time before anything else is done.